### PR TITLE
Update Domain Sender Modal for DMARC record

### PR DESCRIPTION
**Pull request by brezocordero**
## Description

This PR:
a. Slightly modifies the layout of the sender domain modal in preparation for the DMARC record

b. Changes the way the authorized email and DMARC errors are displayed:
When using an unauthorized email address and the domain does not pass the DMARC check we don't display both errors.
We display the unauthorized email address error first and allow to verify the domain once it is authorized.

## Code review notes

To test the layout with 4 records, you could mock the API response or apply this patch:
[patch.zip](https://github.com/mailpoet/mailpoet/files/13691367/patch.zip)

## QA notes

Until the API in the shop is modified you can only do regression tests on the modal to verify a sender domain 

Please test the following scenario on the settings screen or on the Send screen for a newsletter:
Change the sender to a non-authorized email for a domain that has a strict DMARC policy and is not verified (Ex: an alias of your automattic email address)
Check that you only see the error for the email
With an authorized email for a domain that has a strict DMARC policy and is not verified
Check that you see the DMARC error, click on Authorize, check that the modal to verify the domain works as expected

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5770]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage -> See 5796
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5770]: https://mailpoet.atlassian.net/browse/MAILPOET-5770?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/common/manage-sender-domain/manage-sender-domain.tsx [hunk 1]
```diff
@@ -30,23 +30,28 @@ function ManageSenderDomain({
     <div>
       <Heading level={2}> {__('Manage Sender Domain ', 'mailpoet')} </Heading>
       <p>
-        {ReactStringReplace(
+        {__(
+          'Authenticate your sender domain to send emails from your email address. This helps your recipients verify you are the author of these emails and helps mailbox providers fight spam and improves your email delivery rates',
```
Comment (johnolek) [inline mailpoet/assets/js/src/common/manage-sender-domain/manage-sender-domain.tsx]: Missing punctuation
```suggestion
          'Authenticate your sender domain to send emails from your email address. This helps your recipients verify you are the author of these emails and helps mailbox providers fight spam and improves your email delivery rates.',
```

File: mailpoet/assets/js/src/newsletters/send/sender-address-field.jsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { Component } from 'react';
 import PropTypes from 'prop-types';
 import { MailPoet } from 'mailpoet';
 import { FormFieldText } from 'form/fields/text.jsx';
-import { SenderEmailAddressWarning } from 'common/sender-email-address-warning.jsx';
+import { SenderEmailAddressWarning } from 'common/sender-email-address-warning.tsx';
```
Comment (johnolek) [inline mailpoet/assets/js/src/newsletters/send/sender-address-field.jsx]: Not a big deal but I think we can simply drop the extension?

```suggestion
import { SenderEmailAddressWarning } from 'common/sender-email-address-warning';
```

**Review by johnolek — CHANGES_REQUESTED** [review]
@brezocordero Looks good overall! Just a couple of tiny comments. Thanks for including the TS refactor!

**Review by johnolek — APPROVED** [review]

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**